src/replica/duplication/duplication_sync_timer.cpp

```diff
@@ -211,5 +232,14 @@ duplication_sync_timer::get_dup_states(int app_id, /*out*/ bool *app_found)
     return result;
 }
 
+bool duplication_sync_timer::replica_is_cloing_or_closed(gpid id)
+{
+    if (_stub->_closing_replicas.find(id) != _stub->_closing_replicas.end() ||
```

Comment:
Move this function to replica_stub is more meaningful, and it's needed to protected by `_replicas_lock` to avoid data race.

Reply:
I think it's possible to judge the replica's state by itself, maybe by `status()`, it's not needed to access replica_stub.

Reply:
> I think it's possible to judge the replica's state by itself, maybe by `status()`, it's not needed to access replica_stub.

@ninsmiracle take a look at this.